azure-pipelines.yml

```diff
@@ -34,51 +31,77 @@ schedules:
     branches:
       include:
       - main
-      - 1.3
-    always: false
+    always: true
 
-jobs:
-- template: run-tox-env.yml@OpenAstronomy
-  parameters:
-    submodules: false
-    coverage: codecov
-    envs:
-      - macos: py37
-        name: py37_test_macos
+pr:
+  autoCancel: true
 
-      - windows: py38
-        name: py38_test_windows
+stages:
+  - stage: FirstPhaseTests
+    displayName: Core Tests
+    jobs:
+    - template: run-tox-env.yml@OpenAstronomy
+      parameters:
+        default_python: '3.8'
+        submodules: false
+        coverage: codecov
+        toxdeps: tox-pypi-filter
+        envs:
+          - linux: codestyle
+            name: style_check
+            pytest: false
+            libraries: {}
+          - linux: py39
 
-      - linux: py38
-        name: py38_test_linux
+  - stage: SecondPhaseTests
+    displayName: Stage 2 Tests
+    dependsOn: FirstPhaseTests
+    jobs:
+    - template: run-tox-env.yml@OpenAstronomy
+      parameters:
+        default_python: '3.8'
+        submodules: false
+        coverage: codecov
+        toxdeps: tox-pypi-filter
+        libraries:
+          apt:
+            - graphviz
+        envs:
+          - macos: py37
+          - windows: py38
+          - linux: build_docs
+            posargs: " "
+            pytest: false
+          - linux: py38-conda
+            libraries: {}
 
-      - linux: py38-devdeps
-        name: py38_test_devdeps
+  - ${{ if or(eq(variables['Build.Reason'], 'Schedule'), eq(variables['Build.Reason'], 'Manual')) }}:
```

Comment:
Given the lower traffic on this repo and our tight coupling with gwcs I would rather us just run the devdeps build on PRs as well. We could consider re-making the cron when we have released 2.0